### ISSUE TITLE
Bug: --json mode going into a infinite loop?

### ISSUE THREAD

**Issue by hemanth** [ollama/ollama#1182]
```sh
/tmp took 37s 
❯ ollama run llama2 --format json
>>> give me 10 emojis with their meanings
{   

  

 

 

 

 .... # never ends for that input
```

https://github.com/jmorganca/ollama/assets/18315/6771fd1f-d0e3-4f1f-9e7b-e15bacf8acad

^ recording

**Comment by jmorganca** [comment]
Hi @hemanth sorry you hit this issue. It's important sometimes to add `Answer in JSON` in the response so the model can better predict JSON fields (I'll still label this as a bug)

**Comment by hemanth** [comment]
```sh
>>> give me 3 emojies and meanings in JSON format
{
  "emojis":[
    {"name":"😂", "meaning":"laughter"},
    {"name":"🤔", "meaning":"thoughtfulness"},
    {"name":"🎉", "meaning":"celebration"}
  ] 
}
```
Nice!

**Comment by igorschlum** [comment]
For me it's working without specifying json in the prompt. (I have 32Gb on M1Pro MacBook)

(base) igor@macIgor ~ % ollama run llama2 --format json
>>> give me 10 emojis with their meanings
{ "😊": "Happy", "👍": "Okay", "🤔": "Thinking", "🚀": "Space", "🎉": "Celebration", "💭": "Thoughts", "🤝": 
"Friendship", "🌈": "Rainbow", "🏆": "Trophy", "📚": "Book" }

**Comment by nathanleclaire** [comment]
It seems to spin out of control for me quite often in the "REPL" mode specifically. It will just go nutty dumping newlines after it finishes the JSON object.

**Comment by BruceMacD** [comment]
Resolving this to unify with #1910, which will be solved by #3080
